agent/reconciler/reconciler_test.go

```diff
@@ -425,6 +617,63 @@ runCmd:
 				}))
 			})
 
+			Context("When use-installer-controller is set", func() {
+				BeforeEach(func() {
+					hostReconciler.UseInstallerController = true
+				})
+
+				It("should return an error if we fail to load the uninstallation script", func() {
+					byoHost.Spec.UninstallationScript = nil
+					Expect(patchHelper.Patch(ctx, byoHost, patch.WithStatusObservedGeneration{})).NotTo(HaveOccurred())
+
+					result, reconcilerErr := hostReconciler.Reconcile(ctx, controllerruntime.Request{
+						NamespacedName: byoHostLookupKey,
+					})
+					Expect(result).To(Equal(controllerruntime.Result{}))
+					Expect(reconcilerErr).To(MatchError("UninstallationScript not found in Byohost " + byoHost.Name))
+				})
+
+				It("should return error if uninstall script execution failed ", func() {
+					uninstallScript = `testcommand`
+					byoHost.Spec.UninstallationScript = &uninstallScript
+					Expect(patchHelper.Patch(ctx, byoHost, patch.WithStatusObservedGeneration{})).NotTo(HaveOccurred())
+
+					result, reconcilerErr := hostReconciler.Reconcile(ctx, controllerruntime.Request{
+						NamespacedName: byoHostLookupKey,
+					})
+					Expect(result).To(Equal(controllerruntime.Result{}))
+					Expect(reconcilerErr).To(HaveOccurred())
+				})
+
+				FIt("should set K8sComponentsInstallationSucceeded to false if uninstall suceeds", func() {
+					uninstallScript = `echo "uninstall success"`
+					byoHost.Spec.UninstallationScript = &uninstallScript
+					Expect(patchHelper.Patch(ctx, byoHost, patch.WithStatusObservedGeneration{})).NotTo(HaveOccurred())
+
+					result, reconcilerErr := hostReconciler.Reconcile(ctx, controllerruntime.Request{
+						NamespacedName: byoHostLookupKey,
+					})
+					Expect(result).To(Equal(controllerruntime.Result{}))
+					Expect(reconcilerErr).ToNot(HaveOccurred())
+
+					updatedByoHost := &infrastructurev1beta1.ByoHost{}
+					err := k8sClient.Get(ctx, byoHostLookupKey, updatedByoHost)
+					Expect(err).ToNot(HaveOccurred())
+
+					K8sNodeBootstrapSucceeded := conditions.Get(updatedByoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded)
+					Expect(*K8sNodeBootstrapSucceeded).To(conditions.MatchCondition(clusterv1.Condition{
+						Type:     infrastructurev1beta1.K8sNodeBootstrapSucceeded,
+						Status:   corev1.ConditionFalse,
+						Reason:   infrastructurev1beta1.K8sNodeAbsentReason,
+						Severity: clusterv1.ConditionSeverityInfo,
+					}))
+				})
```

Comment:
Missed test case "It should reset `byoHost.Spec.UninstallationScript`  if uninstall succeeds"

Reply:
Also, "It should reset  `byoHost.Spec.InstallationSecret` if uninstall succeeds"

Reply:
added Thanks